### ISSUE TITLE
[Enhancement] Karpenter add-on should allow the creation of a provisioner without requiring all of the fields

### ISSUE THREAD

**Issue by tsahiduek** [aws-quickstart/cdk-eks-blueprints#325]
Per Karpenter [docs](https://karpenter.sh/v0.7.3/provisioner/#specrequirements), all of the [`ProvisionerSpecs`](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/karpenter/index.ts#L16) field are not mandatory/required
The Karpenter add-on should allow the creation of minimal provisioner possible (or default to creation of one)

**Comment by aceat64** [comment]
An example use-case, I have a cluster that I wish to run on spot instances, I don't care about which instance types are used or in what zone, and the default architecture of amd64 is acceptable. Currently I have to create my own Karpenter Provisioner with the requirements set as such:

```yaml
  requirements:
  - key: karpenter.sh/capacity-type
    operator: In
    values:
    - spot
    - on-demand
```

**Comment by youngjeong46** [comment]
@tsahiduek and @aceat64 - this is in progress. It will be a part of PR that will address #317.

**Comment by youngjeong46** [comment]
@tsahiduek @aceat64 The PR is merged, expect this to show up as part of v1.0 release.
